core/src/main/scala/org/labrad/data/Pretty.scala

```diff
@@ -0,0 +1,142 @@
+package org.labrad.data
+
+/**
+ * An implementation of pretty printing based on "A Prettier Printer" by Philip Wadler:
+ * https://homepages.inf.ed.ac.uk/wadler/papers/prettier/prettier.pdf
+ */
+object Pretty {
```

Comment:
I'm not sure how to test this, to be quite honest. I sort of feel like if anything it would be useful to add test cases of formatted labrad data, so that this library can be refactored if needed, but even then it's not like we necessarily want to _preserve_ the formatting exactly as-is, just want it to be parseable, which is what the existing tests check (human-readable is a nice side-benefit...).